library/x509_crt.c

```diff
@@ -101,10 +101,6 @@ typedef struct {
  */
 const mbedtls_x509_crt_profile mbedtls_x509_crt_profile_default =
 {
-#if defined(MBEDTLS_TLS_DEFAULT_ALLOW_SHA1_IN_CERTIFICATES)
```

Comment:
It doesn't matter that much since this file is going to be removed in mbed-crypto, but this commit makes no sense. First, precisely because this file is going to be removed in mbed-crypto. Second, because if this file wasn't going to be removed, this is wrong: it breaks applications that use the (deprecated, but supported) option `MBEDTLS_TLS_DEFAULT_ALLOW_SHA1_IN_CERTIFICATES`, since they will no longer allow SHA-1 in certificates.

It may be time to remove this deprecated option, but that's a decision for the Mbed TLS project, not for the Mbed Crypto project, and it should go together with removing the option from `config.h`, not just breaking one crucial part of it.

Reply:
The commit is present, even if we are removing it, because X.509 and TLS are removed in separate commits, TLS being removed first. We still want to be able to build and have tests pass in intermediate commits, so ether X.509 has to stop depending on TLS before it is removed, or we remove TLS, X.509, and NET all in one commit.

Reply:
I don't see anything wrong with removing TLS and X509 in the same commit. Or removing TLS except for (some of) its configuration, then removing X509, then removing stale configuration options. But I'm against having a major functionality break in the history, as in, a clearly advertised feature of the library still compiles but no longer has the intended effect.

Reply:
OK, we'll move to removing TLS, X509, and NET in one commit since that means less noise like this.